pdns/dnssecinfra.hh

```diff
@@ -37,55 +37,81 @@ class DNSCryptoKeyEngine
     explicit DNSCryptoKeyEngine(unsigned int algorithm) : d_algorithm(algorithm) {}
     virtual ~DNSCryptoKeyEngine() {};
     virtual string getName() const = 0;
-    
+
     typedef std::map<std::string, std::string> stormap_t;
     typedef std::vector<std::pair<std::string, std::string > > storvector_t;
     virtual void create(unsigned int bits)=0;
+    virtual auto createFromPEMFile(DNSKEYRecordContent& drc, const std::string& filename, std::FILE* fp) -> void
```

Comment:
Perhaps a reference to a `std::FILE` would make more sense here, in `makeFromPEMFile` and in `convertToPEM``, since it cannot be `nullptr`?
```suggestion
    virtual auto createFromPEMFile(DNSKEYRecordContent& drc, const std::string& filename, std::FILE& fp) -> void
```